var/spack/repos/builtin/packages/py-fastfold/package.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyFastfold(PythonPackage):
+    """Optimizing Protein Structure Prediction Model Training and Inference on
+    GPU Clusters."""
+
+    homepage = "https://github.com/hpcaitech/FastFold"
+    url = "https://github.com/hpcaitech/FastFold/archive/refs/tags/0.2.0.tar.gz"
+
+    maintainers = ["meyersbs"]
+
+    version("0.2.0", sha256="6760dbae9809b8b26219c9477489d34325807be504098901d0375fbdc3103f88")
+
+    # From README:
+    depends_on("python@3.8:", type=("build", "run"))  # Req Relaxed (was @3.8:3.9)
+    depends_on("cuda@11.1:", type=("build", "run"))
+    depends_on("py-torch@1.10:+cuda", type=("build", "run"))
+    # Needed or py-torch fails to build when python@3.10:
+    depends_on("cudnn@8.0.5:", type=("build", "run"), when="python@3.10:")
```

Comment:
This should go in the py-torch package then. Can you open a separate PR for that and remove it here? A link to the issue or discussion of why this is necessary in the PR would be helpful, it isn't immediately clear to me why this would matter.

Reply:
I'm not clear on why it matters, either. When I did `spack install py-fastfold ^python@3.10:` without that line, spack grabbed `cudnn@8.0.4.30-11.1` and the py-torch install exploded. I'll talk to @qwertos and see if we can debug further.

Reply:
I'm working on a fix for this, I'll ping you when it's ready.

Reply:
Fixed in #34478